src/components/app.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from "react";
+import {
+  FF_Sphere,
+  FF_Rastrigin,
+  FF_Schwefel,
+  FF_2D
+} from "../service/optimisation-functions/functions";
+import PSO from "../service/pso/pso";
+import Particle from "../service/pso/particle";
+import FunctionPlotter3D from "./plotters/plot3D";
+import FunctionPlotter2D from "./plotters/plot2D";
+import TopParticles from "./top-particles/topParticles";
+
+const Gun = require("gun/gun");
+require("gun/lib/not.js");
+require("gun/sea");
+
+import { eventBus } from "../event-bus/eventBus";
+
+import "./app.css";
+
+const TIME_BETWEEN_ITERATIONS = 120;
+
+const App = () => {
+  const [pso, setPSO] = useState(null);
+  const [gun] = useState(Gun(location.origin + "/gun"));
+
+  function initializePopulation() {
+    const numParticles = 10;
+    let particles = [];
+    const fitnessFunction = new FF_Schwefel();
+
+    for (let i = 0; i < numParticles; i++) {
+      const uniqueId = particles.length;
+      let p = new Particle(fitnessFunction, uniqueId);
+      particles.push(p);
+    }
+
+    const pso = new PSO(fitnessFunction, particles);
+    gun.get("global-minimum").not(function(key) {
```

Comment:
GUN should keep global mimima for multiple functions `gun.get("rastrigin-global-minima")` so you can execute multiple optimisations on the same database